Neos.Flow/Tests/Unit/Mvc/Routing/Dto/RouteLifetimeTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace Neos\Flow\Tests\Unit\Mvc\Routing\Dto;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Mvc\Routing\Dto\RouteLifetime;
+use Neos\Flow\Tests\UnitTestCase;
+
+/**
+ * Testcase for the RouteLifetime DTO
+ */
+class RouteLifetimeTest extends UnitTestCase
+{
+
+    /**
+     * @test
+     */
+    public function createFromNegativeIntegerThrowsInvalidArgumentException()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        RouteLifetime::createFromInt(-1);
+    }
+
+    /**
+     * @test
+     */
+    public function createFromIntCreatesANewInstanceWithTheGivenValue()
+    {
+        $lifetime = RouteLifetime::createFromInt(123);
+        self::assertSame(123, $lifetime->getValue());
+    }
+
+    /**
+     * @test
+     */
+    public function createEmptyCreatesANewInstanceWithNullValue()
+    {
+        $lifetime = RouteLifetime::createUndefined();
+        self::assertSame(null, $lifetime->getValue());
+    }
+
+    public function mergeReturnsLowerLifetimeOfNonNullValuesDataProvider(): array
+    {
+        return [
+            [100, 200, 100],
+            [100, 100, 100],
+            [null, 200, 200],
+            [200, null, 200],
+            [null, null, null]
```

Comment:
This needs a case that contains `0` IMO since that seems to have a special meaning

Reply:
You are right ... added more cases